src/operations/query/filters/contains.js

```diff
@@ -6,30 +6,29 @@
  */
 const {escapeRegExp} = require('../../../utils/regexEscaper');
 
-function paramMatch(fields, name) {
-    return fields.find((field) => field === name);
-}
-
 /**
  * filters by contains
  * https://www.hl7.org/fhir/search.html#string
  * @param {import('../../common/types').SearchParameterDefinition} propertyObj
- * @param {string} queryParameter
- * @param {Object} args
+ * @param {string} queryParameterValue
  * @param {Set} columns
  * @return {Object[]}
  */
-function filterByContains({propertyObj, queryParameter, args, columns}) {
+function filterByContains({propertyObj, queryParameterValue, columns}) {
     /**
      * @type {Object[]}
      */
     const and_segments = [];
     and_segments.push({
-        [`${propertyObj.field || paramMatch(propertyObj.fields, queryParameter)}`]:
-            {
-                $regex: escapeRegExp(args[`${queryParameter}:contains`]),
-                $options: 'i',
-            },
+        '$or': Array.from(columns).map(c => {
```

Comment:
The idea here (which is also applied to the above and below filters) is that search on some data types, such as codes, results in different fields being searched, not just "code". So we'll look for the regex to be true in any one of those other fields, based on whichever columns were marked as being relevant to this query parameter by the previous datatype-based logic in r4